File: Glide distance estimator .py
glide_ratio = {
    "aircraft a": 8,          
    "aircraft b": 15,
    "aircraft c": 9,
    "aircraft d": 17,
    "aircraft e": 7,
    "aircraft f": 16,
    "aircraft g": 10,
    "aircraft h": 6,
    "aircraft i": 14,
    "aircraft j": 6,
    "aircraft k": 8,
    "aircraft l": 5,
    "aircraft m": 14,
    "aircraft n": 18
}   

def ind():
    print('''
                aircraft a      
                aircraft b       
                aircraft c      
                aircraft d      
                aircraft e      
                aircraft f      
                aircraft g      
                aircraft h
                aircraft i
                aircraft j     
                aircraft k
                aircraft l
                aircraft m
                aircraft n        ''')
                                  

    

ind()
print("Enter a aircraft from index and a altitude to get the estimated glide distance.")
print("Enter i for index and x to exit.")
while True:
    try:
        print("Enter a aircraft in index.")
        search = input()
        if search == "i":
            ind()
            continue
        elif search == "x":
            break
        print("Enter a altitude")
        altitude = input()
        if altitude == "i":
            ind()
            continue
        elif altitude == "x":
            break
        print(f"\nThe glide distance is {float(altitude) *  float(glide_ratio[search]) / 5280} miles.\n") 
    except KeyError:
        print("Enter a valid aircraft name.")
        print("Try again.")
    except ValueError:   
        print("Enter a valid altitude.")
        print("Try again.")       
        
        
 
      
    


        
    

File: Glide distance estimator.cpp
//WARNING: Educational demo only - NOT FOR USE IN REAL-WORLD AVIATION.
//PROVIDED "AS IS" with no warranty.

#include <iostream>
#include <map>
#include <string>

void checkInput() {
    std::cerr << "\nInvalid input.\n\n";
    std::cin.clear();
}

void ind() {
    using namespace std;
    cout << "               \n";
    cout << "     aircraft a\n";
    cout << "     aircraft b\n";
    cout << "     aircraft c\n";
    cout << "     aircraft d\n";
    cout << "     aircraft e\n";
    cout << "     aircraft f\n";
    cout << "     aircraft g\n";
    cout << "               \n";
}

int main() {
    std::ios_base::sync_with_stdio(false);
    std::map<std::string, double> gr = {         
        {"aircraft a", 15},
        {"aircraft b",  9},
        {"aircraft c", 17},
        {"aircraft d",  7},
        {"aircraft e", 10},
        {"aircraft f", 17},
        {"aircraft g", 15}
    };
        
    std::string search;
    std::string altitude;

    ind();
    std::cout << "Enter a aircraft from index and a altitude to get the estimated glide distance.\n";
    std::cout << "Enter i for index and x to exit.\n";
    while (true) {
        try {
            std::cout << "Enter a aircraft in index.\n";
            while(!getline(std::cin, search)) {
                checkInput();
            }
            if (search == "x") {
                break;
            } else if (search == "i") {
                ind();
                continue;
            }
            std::cout << "Enter a altitude.\n";
            while(!getline(std::cin, altitude)) {
                checkInput();
            }
            if (altitude == "i") {
                ind();
                continue;
            } else if (altitude == "x") {
                break;
            }
            double result = stod(altitude) * gr.at(search) / 5280;
            std::cout << "\nThe estimated glide distance is " << result << " miles.\n\n";
        } catch (std::out_of_range) {
            std::cout << "Enter a aircraft in index.\n\n";
        } catch (std::invalid_argument) {
            std::cout << "Enter a valid altitude.\n\n";
        }
    }

    return 0;
}

